core/src/test/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSupervisorSpec.scala

```diff
@@ -0,0 +1,147 @@
+package org.broadinstitute.dsde.rawls.jobexec
+
+import java.util.UUID
+
+import akka.actor.{ActorRef, ActorSystem, PoisonPill}
+import akka.testkit.TestKit
+import org.broadinstitute.dsde.rawls.RawlsTestUtils
+import org.broadinstitute.dsde.rawls.dataaccess.{MockExecutionServiceDAO, MockGoogleServicesDAO, MockShardedExecutionServiceCluster}
+import org.broadinstitute.dsde.rawls.dataaccess.slick.TestDriverComponent
+import org.broadinstitute.dsde.rawls.jobexec.SubmissionSupervisor.{SaveCurrentWorkflowStatusCounts, SubmissionStarted, UpdateGlobalJobExecGauges}
+import org.broadinstitute.dsde.rawls.metrics.RawlsStatsDTestUtils
+import org.broadinstitute.dsde.rawls.model.{SubmissionStatuses, WorkflowStatuses, WorkspaceName}
+import org.broadinstitute.dsde.rawls.util.MockitoTestUtils
+import org.scalatest.{BeforeAndAfterAll, FlatSpecLike, Matchers}
+import org.scalatest.concurrent.Eventually
+
+import scala.concurrent.duration._
+
+class SubmissionSupervisorSpec(_system: ActorSystem) extends TestKit(_system) with FlatSpecLike with Matchers with TestDriverComponent with BeforeAndAfterAll with Eventually with RawlsTestUtils with MockitoTestUtils with RawlsStatsDTestUtils {
+
+  import driver.api._
+
+  def this() = this(ActorSystem("SubmissionSupervisorSpec"))
```

Comment:
Not a big deal, but I wonder why we define this extra constructor instead of:
```
class SubmissionSupervisorSpec extends TestKit(ActorSystem("SubmissionSupervisorSpec")) with Foo with Bar {
  ... 
}
```